pyrometheus/codegen/python.py

```diff
@@ -423,48 +456,62 @@ def get_falloff_rates(self, temperature, concentrations, k_fwd):
         %endfor
         ])*reduced_pressure/(1+reduced_pressure)
 
-        %for j, (i, react) in enumerate(falloff_reactions):
-        k_fwd[${i}] = k_high[${j}]*falloff_function[${j}]*ones
-        %endfor
-        return
+        return k_high*falloff_function
 
     %endif
     def get_fwd_rate_coefficients(self, temperature, concentrations):
         ones = self._pyro_zeros_like(temperature) + 1.0
+        %if falloff_reactions:
+        k_falloff = self.get_falloff_rates(temperature, concentrations)
+        %endif
         k_fwd = [
-        %for react in sol.reactions():
-        %if react.equation in [r.equation for _, r in falloff_reactions]:
-            0*temperature,
+        %for i, react in enumerate(sol.reactions()):
+        %if react.reaction_type.startswith("falloff"):
+        %for k, (j, react) in enumerate(falloff_reactions):
+        %if i == j:
```

Comment:
Is there a way of doing this without nested loops?

Reply:
JAX complains of indexations. So modifying only the reactions that are falloff or three-body individually AFTER the pyro_array expression makes JAX fail.

I won;t deny that this is ugly, we can potentially do it nicer, but currently it works.

Reply:
Agreed on Jax, I actually prefer in general what you have done here to avoid `k_fwd[${I}]` (or whatever it was) in `get_falloff_rates`. However, my concern is the template syntax, specifically the nested `%for i, ...` and `%for k, ...`, along with the `%if i == j`. Let me think how to simplify.